### PR TITLE
make spacemacs/delete-file to actually work on directories

### PR DESCRIPTION
**Pull request by d12frosted**
It didn't really worked on directories, though documentation stated that it should! I couldn't live with such knowledge as a burden, so I tried to fix that. 😸 

P. S. Also fixed that part where it kills buffer before asking user. 

P. P. S. I have implemented helper function that might be used in future. By me at least.

### CODE REVIEW

**Comment by bmag** [comment]
Wait, there really isn't a built-in or idiomatic way to delete a directory recursively and kill all the relevant buffers? How does Dired handle it?

File: layers/+distributions/spacemacs-base/funcs.el [hunk 1]
```diff
@@ -380,16 +380,22 @@ projectile cache when it's possible.
...
-      (delete-file filename)
+  (let ((is-dir (file-directory-p filename)))
+    (when (and (file-exists-p filename)
+               (or (not ask-user)
+                   (yes-or-no-p
+                    (message  "Are you sure you want to delete this %s? "
+                              (if is-dir "directory" "file")))))
+      (spacemacs/kill-file-or-dir-buffers filename)
+      (if (file-directory-p filename)
+          (delete-directory filename t)
```
Comment (bmag) [inline layers/+distributions/spacemacs-base/funcs.el]: Shouldn't we pass also a non-nil `trash` argument?

Comment (d12frosted) [inline layers/+distributions/spacemacs-base/funcs.el]: No why? We don't move files to trash, we remove them, so it's consistent with a line after. 

Though we might add an argument for trashing.

Comment (bmag) [inline layers/+distributions/spacemacs-base/funcs.el]: Well, we delete the files either way. The question is whether to delete them permanently (as `rm` does) or put them in the trash can (as file managers do). If we move the files to the trash then the user can still restore them if he/she realizes it was a mistake to delete the files. I think that should be the default, and that's also the default behavior of `delete-directory` when called interactively (but only when called interactively).

Same comment goes for `delete-file` below, I just missed it before.

Comment (bmag) [inline layers/+distributions/spacemacs-base/funcs.el]: Lets add an optional argument in any case, and lets make it have the reverse meaning when `spacemacs/delete-file` is called interactively. (same as `delete-directory`)

**Comment by Miciah** [comment]
Counsel upstream recently added a "delete" action that uses `dired-delete-file`: https://github.com/abo-abo/swiper/commit/e0293f23fa3d2d027403f826175b301d0d529975.  Do we still need `spacemacs/delete-file`?

**Pull request closed without merging**